include/envoy/stats/stats.h

```diff
@@ -32,32 +34,34 @@ class Metric {
   virtual std::string name() const PURE;
 
   /**
-   * Returns the full name of the Metric as a nul-terminated string. The
-   * intention is use this as a hash-map key, so that the stat name storage
-   * is not duplicated in every map. You cannot use name() above for this,
-   * as it returns a std::string by value, as not all stat implementations
-   * contain the name as a std::string.
-   *
-   * Note that in the future, the plan is to replace this method with one that
-   * returns a reference to a symbolized representation of the elaborated string
-   * (see source/common/stats/symbol_table_impl.h).
+   * Returns the full name of the Metric as an encoded array of symbols.
    */
-  virtual const char* nameCStr() const PURE;
+  virtual StatName statName() const PURE;
 
   /**
    * Returns a vector of configurable tags to identify this Metric.
    */
-  virtual const std::vector<Tag>& tags() const PURE;
+  virtual std::vector<Tag> tags() const PURE;
 
   /**
-   * Returns the name of the Metric with the portions designated as tags removed.
+   * Returns the name of the Metric with the portions designated as tags
+   * removed as a string.
```

Comment:
Could you provide an inline example here and in the docstring for `tagExtractedStatName()`?